deployment/src/main/java/io/quarkiverse/jackson/jq/deployment/JacksonJqProcessor.java

```diff
@@ -1,59 +1,96 @@
+import static io.quarkiverse.jackson.jq.deployment.JacksonJqSupport.lookupFunctionsFromConfig;
+
+import java.util.Collections;
 import java.util.List;
-import java.util.Map;
 
 import javax.inject.Singleton;
 
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.jandex.IndexView;
+
 import io.quarkiverse.jackson.jq.JacksonJqScopeRecorder;
+import io.quarkiverse.jackson.jq.JqFunction;
 import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.AdditionalApplicationArchiveMarkerBuildItem;
+import io.quarkus.deployment.builditem.ApplicationArchivesBuildItem;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
-import io.quarkus.deployment.builditem.nativeimage.NativeImageResourcePatternsBuildItem;
-import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.recording.RecorderContext;
+import io.quarkus.runtime.RuntimeValue;
+import net.thisptr.jackson.jq.BuiltinFunction;
 import net.thisptr.jackson.jq.BuiltinFunctionLoader;
-import net.thisptr.jackson.jq.Function;
 import net.thisptr.jackson.jq.Scope;
-import net.thisptr.jackson.jq.Versions;
 
 class JacksonJqProcessor {
-
     private static final String FEATURE = "jackson-jq";
-    private static final String REFLECTION_VERSION_RANGE_DESERIALIZER = "net.thisptr.jackson.jq.internal.misc.VersionRangeDeserializer";
-    private static final String JSON_CONFIG_GLOB = "net/thisptr/jackson/**/*.json";
 
     @BuildStep
     FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
     @BuildStep
-    List<ReflectiveClassBuildItem> registerForReflection() {
-        final List<ReflectiveClassBuildItem> reflectionConfig = new ArrayList<>();
-        reflectionConfig.add(ReflectiveClassBuildItem.weakClass(REFLECTION_VERSION_RANGE_DESERIALIZER));
-        return reflectionConfig;
-    }
+    @Record(ExecutionTime.STATIC_INIT)
+    SyntheticBeanBuildItem quarkusScopeBean(
+            CombinedIndexBuildItem combinedIndex,
+            ApplicationArchivesBuildItem archives,
+            JacksonJqConfig config,
+            JacksonJqScopeRecorder recorder,
+            RecorderContext context) throws Exception {
 
-    @BuildStep
-    NativeImageResourcePatternsBuildItem includeJsonConfigFile() {
-        return NativeImageResourcePatternsBuildItem.builder().includeGlob(JSON_CONFIG_GLOB).build();
-    }
+        IndexView indexView = combinedIndex.getIndex();
+        List<String> excludes = config.functions.excludes.orElseGet(Collections::emptyList);
+
+        RuntimeValue<Scope> root = recorder.createScope();
+        RuntimeValue<Scope> local = recorder.createScope(root);
```

Comment:
Here we are creating a parent/child scope so the root scope contains the default functions whereas the local, the one that is exposed as a CDI bean, contains the local/custom functions.